Symfony/CS/Tokenizer/Transformer/BraceClassInstantiation.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Tokenizer\Transformer;
+
+use Symfony\CS\Tokenizer\AbstractTransformer;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * Transform braced class instantiation braces in `(new Foo())` into CT_BRACE_CLASS_INSTANTIATION_OPEN
+ * and CT_BRACE_CLASS_INSTANTIATION_CLOSE.
+ *
+ * @author Sebastiaans Stok <s.stok@rollerscapes.net>
+ */
+class BraceClassInstantiation extends AbstractTransformer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(Tokens $tokens)
+    {
+        for ($index = $tokens->count() - 1; 0 <= $index; --$index) {
+            if (!$tokens[$index]->equals('(')) {
+                continue;
+            }
+
+            if (!$tokens[$tokens->getNextMeaningfulToken($index)]->equals(array(T_NEW))) {
+                continue;
+            }
+
+            $openIndex = $index;
+            $closeIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $openIndex);
+
+            if ($tokens[$tokens->getNextMeaningfulToken($closeIndex)]->equalsAny(array(array(T_OBJECT_OPERATOR), array(T_DOUBLE_COLON)))) {
+                $tokens[$openIndex]->override(array(CT_BRACE_CLASS_INSTANTIATION_OPEN, '(', $tokens[$openIndex]->getLine()));
```

Comment:
I'm pretty sure that will work better yes :')

![635412533686232669](https://cloud.githubusercontent.com/assets/904790/6096659/85312342-af9e-11e4-9501-abeb84a299d6.png)